test/BinaryParse.cpp

```diff
@@ -435,15 +341,14 @@ TEST_F(BinaryParseTest, ExtendedInstruction) {
   };
   const auto instruction = MakeInstruction(
       SpvOpExtInst, {2, 3, 1, uint32_t(OpenCLLIB::Entrypoints::Sqrt), 4});
-  const spv_parsed_instruction_t parsed_inst = {instruction.data(),
-                                                uint16_t(instruction.size()),
-                                                SpvOpExtInst,
-                                                SPV_EXT_INST_TYPE_OPENCL_STD,
-                                                2,  // type id
-                                                3,  // result id
-                                                operands.data(),
-                                                uint16_t(operands.size())};
-  EXPECT_THAT(instructions(), ElementsAre(_, ParsedInstruction(parsed_inst)));
+  EXPECT_CALL(client_, Instruction(ParsedInstruction(spv_parsed_instruction_t{
+                           instruction.data(), uint16_t(instruction.size()),
```

Comment:
We've actually known for a while that the style guide discourages constructor casts (to set apart conversions from bitwise casting).  It's just the [rule](https://google.github.io/styleguide/cppguide.html#Casting)'s awkward wording that makes it sound like a C cast.

In the spirit of adopting the Google style guide over time, I've changed all the constructor casts to `static_cast`.